dwave/cloud/solver.py

```diff
@@ -1043,19 +1063,62 @@ def _sample(self, type_, linear, quadratic, offset, params,
 
         return computation
 
+    # kept for internal backwards compatibility and in case it's being
+    # used externally anywhere.
     def _format_params(self, type_, params):
         """Reformat some of the parameters for sapi."""
-        if 'initial_state' in params:
-            # NB: at this moment the error raised when initial_state does not match lin/quad (in
-            # active qubits) is not very informative, but there is also no clean way to check here
-            # that they match because lin can be either a list or a dict. In the future it would be
-            # good to check.
-            initial_state = params['initial_state']
-            if isinstance(initial_state, Mapping):
+        self.reformat_parameters(type_, params, self.properties, inplace=True)
+
+    @staticmethod
+    def reformat_parameters(vartype: '_Vartype',
+                            parameters: typing.MutableMapping[str, typing.Any],
+                            properties: typing.Mapping[str, typing.Any],
+                            inplace: bool = False,
+                            ) -> typing.MutableMapping[str, typing.Any]:
+        """Reformat some solver parameters for SAPI.
 
-                initial_state_list = [3]*self.properties['num_qubits']
+        Currently the only reformatted parameter is ``initial_state``. This
+        method allows ``initial_state`` to be submitted as a dictionary
+        mapping the qubits to their initial value.
 
-                low = -1 if type_ == 'ising' else 0
+        Args:
+            vartype: One of ``'ising'`` or ``'qubo'``. If :mod:`dimod` is
+                installed, this can also be any
+                :class:`~dimod.typing.VartypeLike`.
+            parameters: The parameters to submit to ths solver.
+            properties: The solver's properties. Note that this will
+                work with either :attr:`StructuredSolver.properties`
+                or :attr:`dwave.systems.DWaveSampler.properties`.
```

Comment:
```suggestion
            properties: The solver's properties. Supported values 
                are :attr:`StructuredSolver.properties`,
                :attr:`dwave.systems.DWaveSampler.properties`,
                or similar.
```

Reply:
I don't think that's actually very clear. I could say ":attr:`StructuredSolver.properties` ... can be provided."

Reply:
There's an ambiguity with the "will work with either": do you mean it has to be one of those or that it can be one of those. I understood the latter

Reply:
It can be either, or other options as well.

Reply:
How about that (edited)?

Reply:
My concern is more with "can be set to" as applied to an attribute, a mutable one at that. I will perhaps just drop the clause about the properties altogether and move that to an examples section.

Reply:
Updated